core/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -914,6 +917,36 @@ public void testRecoverFromStore() throws IOException {
         closeShards(newShard);
     }
 
+    public void testRecoverFromStoreWithOutOfOrderDelete() throws IOException {
+        final IndexShard shard = newStartedShard(false);
+        int translogOps = 1;
```

Comment:
I think you can make this `final` by declaring it immediately before the random flush choice and initializing it either there or in an else branch to 0 or 1, respectively, depending on the outcome.